charts/helm-chart/kubernetes-dashboard/values.yaml

```diff
@@ -12,288 +12,286 @@
-#   value: 'some value'
-
-## Additional volumes to be added to kubernetes dashboard pods
-##
-extraVolumes: []
-# - name: dashboard-kubeconfig
-#   secret:
-#     defaultMode: 420
-#     secretName: dashboard-kubeconfig
-
-## Additional volumeMounts to be added to kubernetes dashboard container
-##
-extraVolumeMounts: []
-# - mountPath: /kubeconfig
-#   name: dashboard-kubeconfig
-#   readOnly: true
-
-## Array of extra K8s manifests to deploy
-##
-extraManifests: []
-# - apiVersion: v1
-#   kind: ConfigMap
-#   metadata:
-#     name: additional-configmap
-#   data:
-#     mykey: myvalue
-
-## Annotations to be added to kubernetes dashboard pods
-# podAnnotations:
-
-## SecurityContext to be added to kubernetes dashboard pods
-## To disable set the following configuration to null:
-# securityContext: null
-securityContext:
-  runAsNonRoot: true
-  seccompProfile:
-    type: RuntimeDefault
-
-## SecurityContext defaults for the kubernetes dashboard container and metrics scraper container
-## To disable set the following configuration to null:
-# containerSecurityContext: null
-containerSecurityContext:
-  allowPrivilegeEscalation: false
-  readOnlyRootFilesystem: true
-  runAsUser: 1001
-  runAsGroup: 2001
-  capabilities:
-    drop: ["ALL"]
-
-## @param podLabels Extra labels for OAuth2 Proxy pods
-## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
-##
-podLabels: {}
-## @param podAnnotations Annotations for OAuth2 Proxy pods
-## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
-##
-podAnnotations: {}
-
-## Node labels for pod assignment
-## Ref: https://kubernetes.io/docs/user-guide/node-selection/
-##
-nodeSelector: {}
-
-## List of node taints to tolerate (requires Kubernetes >= 1.6)
-tolerations: []
-#  - key: "key"
-#    operator: "Equal|Exists"
-#    value: "value"
-#    effect: "NoSchedule|PreferNoSchedule|NoExecute"
-
-## Affinity for pod assignment
-## ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
-affinity: {}
-
-## Name of Priority Class of pods
-# priorityClassName: ""
-
-## Pod resource requests & limits
-resources:
-  requests:
-    cpu: 100m
-    memory: 200Mi
-  limits:
-    cpu: 2
-    memory: 200Mi
-
-## Serve application over HTTP without TLS
-##
-## Note: If set to true, you may want to add --enable-insecure-login to extraArgs
-protocolHttp: false
-
-service:
-  type: ClusterIP
-  # Dashboard service port
-  externalPort: 443
-
-  ## LoadBalancerSourcesRange is a list of allowed CIDR values, which are combined with ServicePort to
-  ## set allowed inbound rules on the security group assigned to the master load balancer
-  # loadBalancerSourceRanges: []
-
-  # clusterIP: ""
-
-  ## A user-specified IP address for load balancer to use as External IP (if supported)
-  # loadBalancerIP:
-
-  ## Additional Kubernetes Dashboard Service annotations
+# General configuration shared across resources
+app:
```

Comment:
Aren't there some use cases where you want to differentiate some values between the api and the components? i.e should we avoid too much intelligence in here and just have two blocks?

Reply:
I have tried to extract only the parts that should be the same across all components. There still might be some room for improvement, but we will indeed have more and more microservices in the future. A generic configuration for security and image handling might be a good idea.